File: Core/src/main/java/com/alphaz/core/authorization/entity/OauthRefreshTokenEntity.java
package com.alphaz.core.authorization.entity;

import javax.persistence.*;
import java.util.Arrays;
import java.util.Objects;

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_refresh_token", catalog = "")
public class OauthRefreshTokenEntity {
    private String tokenId;
    private byte[] token;
    private byte[] authentication;
    private Long id;

    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    @Basic
    @Column(name = "token_id")
    public String getTokenId() {
        return tokenId;
    }

    public void setTokenId(String tokenId) {
        this.tokenId = tokenId;
    }

    @Basic
    @Column(name = "token")
    public byte[] getToken() {
        return token;
    }

    public void setToken(byte[] token) {
        this.token = token;
    }

    @Basic
    @Column(name = "authentication")
    public byte[] getAuthentication() {
        return authentication;
    }

    public void setAuthentication(byte[] authentication) {
        this.authentication = authentication;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OauthRefreshTokenEntity that = (OauthRefreshTokenEntity) o;
        return Objects.equals(tokenId, that.tokenId) &&
                Arrays.equals(token, that.token) &&
                Arrays.equals(authentication, that.authentication);
    }

    @Override
    public int hashCode() {

        int result = Objects.hash(tokenId);
        result = 31 * result + Arrays.hashCode(token);
        result = 31 * result + Arrays.hashCode(authentication);
        return result;
    }
}

File: Core/src/main/kotlin/com/alphaz/core/authorization/auth2/OauthRefreshTokenEntity.kt
package com.alphaz.core.authorization.auth2

import java.util.*
import javax.persistence.*

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_refresh_token", schema = "", catalog = "")
class OauthRefreshTokenEntity {
    @get:Basic
    @get:Column(name = "token_id")
    var tokenId: String? = null
    @get:Basic
    @get:Column(name = "token")
    var token: ByteArray? = null
    @get:Basic
    @get:Column(name = "authentication")
    var authentication: ByteArray? = null
    @get:javax.persistence.Id
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null


}
